components/camel-jq/src/main/java/org/apache/camel/language/jq/JqExpression.java

```diff
@@ -0,0 +1,244 @@
+package org.apache.camel.language.jq;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.apache.camel.CamelContext;
+import org.apache.camel.Exchange;
+import org.apache.camel.RuntimeCamelException;
+import org.apache.camel.spi.ExpressionResultTypeAware;
+import org.apache.camel.support.CamelContextHelper;
+import org.apache.camel.support.ExpressionAdapter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.BooleanNode;
+
+import net.thisptr.jackson.jq.BuiltinFunctionLoader;
+import net.thisptr.jackson.jq.Function;
+import net.thisptr.jackson.jq.JsonQuery;
+import net.thisptr.jackson.jq.Scope;
+import net.thisptr.jackson.jq.Versions;
+import net.thisptr.jackson.jq.exception.JsonQueryException;
+
+public class JqExpression extends ExpressionAdapter implements ExpressionResultTypeAware {
+    private static final Logger LOGGER = LoggerFactory.getLogger(JqExpression.class);
+
+    private final String expression;
+    private final Scope scope;
+
+    private ObjectMapper objectMapper;
+    private String resultTypeName;
+    private Class<?> resultType;
+
+    private boolean autoDiscoverObjectMapper;
```

Comment:
I wonder if you want/need for these fine grained auto discover functions to be able to turn them on|off individually?

Would you for example turn off fromJq only? Or from serviceLoader only? And are these two built-in functions from JQ itself?